AVX512 does not compile with g++

xxhash passes `int` where `_MM_PERM_ENUM` is required, for example:

```
../../../server/tracy_xxh3.h: In function ‘void XXH3_accumulate_512_avx512(void*, const void*, const void*)’:
../../../server/tracy_xxh3.h:1000:69: error: invalid conversion from ‘int’ to ‘_MM_PERM_ENUM’ [-fpermissive]
 1000 |         __m512i const data_key_lo = _mm512_shuffle_epi32 (data_key, _MM_SHUFFLE(0, 3, 0, 1));
      |                                                                     ^~~~~~~~~~~
      |                                                                     |
      |                                                                     int
In file included from /usr/lib/gcc/x86_64-linux-gnu/9/include/immintrin.h:55,
                 from ../../../server/tracy_xxh3.h:76:
/usr/lib/gcc/x86_64-linux-gnu/9/include/avx512fintrin.h:4364:50: note:   initializing argument 2 of ‘__m512i _mm512_shuffle_epi32(__m512i, _MM_PERM_ENUM)’
 4364 | _mm512_shuffle_epi32 (__m512i __A, _MM_PERM_ENUM __mask)
      |                                    ~~~~~~~~~~~~~~^~~~~~
```

Fix seems simple enough: https://github.com/embree/embree/commit/40b9aca2668f443cae6bfbfa7cc5a354f1087011

Relevant discussion: https://developercommunity.visualstudio.com/content/problem/107147/avx512-permute-macros-are-missing.html

--- a/.travis.yml
+++ b/.travis.yml
@@ -22,6 +22,8 @@ matrix:
         - make -B test-all
         - make clean
         - make dispatch
+        - make clean
+        - CC=g++ make CFLAGS="-O2 -mavx512f"
 
     - name: Check results consistency on x64
       arch: amd64

--- a/.travis.yml
+++ b/.travis.yml
@@ -23,7 +23,7 @@ matrix:
         - make clean
         - make dispatch
         - make clean
-        - CC=g++ make CFLAGS="-O2 -mavx512f"
+        - CC=g++ CFLAGS="-O1 -mavx512f" make
 
     - name: Check results consistency on x64
       arch: amd64

--- a/xxh3.h
+++ b/xxh3.h
@@ -997,11 +997,11 @@ XXH3_accumulate_512_avx512(void* XXH_RESTRICT acc,
         /* data_key    = data_vec ^ key_vec; */
         __m512i const data_key    = _mm512_xor_si512     (data_vec, key_vec);
         /* data_key_lo = data_key >> 32; */
-        __m512i const data_key_lo = _mm512_shuffle_epi32 (data_key, _MM_SHUFFLE(0, 3, 0, 1));
+        __m512i const data_key_lo = _mm512_shuffle_epi32 (data_key, (_MM_PERM_ENUM)_MM_SHUFFLE(0, 3, 0, 1));
         /* product     = (data_key & 0xffffffff) * (data_key_lo & 0xffffffff); */
         __m512i const product     = _mm512_mul_epu32     (data_key, data_key_lo);
         /* xacc[0] += swap(data_vec); */
-        __m512i const data_swap = _mm512_shuffle_epi32(data_vec, _MM_SHUFFLE(1, 0, 3, 2));
+        __m512i const data_swap = _mm512_shuffle_epi32(data_vec, (_MM_PERM_ENUM)_MM_SHUFFLE(1, 0, 3, 2));
         __m512i const sum       = _mm512_add_epi64(*xacc, data_swap);
         /* xacc[0] += product; */
         *xacc = _mm512_add_epi64(product, sum);
@@ -1046,7 +1046,7 @@ XXH3_scrambleAcc_avx512(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
         __m512i const data_key    = _mm512_xor_si512     (data_vec, key_vec);
 
         /* xacc[0] *= XXH_PRIME32_1; */
-        __m512i const data_key_hi = _mm512_shuffle_epi32 (data_key, _MM_SHUFFLE(0, 3, 0, 1));
+        __m512i const data_key_hi = _mm512_shuffle_epi32 (data_key, (_MM_PERM_ENUM)_MM_SHUFFLE(0, 3, 0, 1));
         __m512i const prod_lo     = _mm512_mul_epu32     (data_key, prime32);
         __m512i const prod_hi     = _mm512_mul_epu32     (data_key_hi, prime32);
         *xacc = _mm512_add_epi64(prod_lo, _mm512_slli_epi64(prod_hi, 32));

